### PR TITLE
Set default value for provisioned key in rbd sync

### PR DESCRIPTION
**Pull request by anmolsachan**
Though the issue could not be replicated, for a safe condition we can set the provisioned value to be None if it is not found during sync.

tendrl_issue_id: https://github.com/Tendrl/ceph-integration/issues/172
Signed-off-by: anmolsachan asachan@redhat.com

### CODE REVIEW

**Comment by anmolsachan** [comment]
@shtripat @GowthamShanmugam please review

File: tendrl/ceph_integration/sds_sync/__init__.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def _sync_rbds(self):
                     size=v['size'],
                     pool_id=pool_id,
                     flags=v['flags'],
-                    provisioned=self._to_bytes(v['provisioned']),
+                    provisioned=self._to_bytes(v.get("provisioned", 0)),
```
Comment (cybernth) [inline tendrl/ceph_integration/sds_sync/__init__.py]: do you really want to set the value to 0 if not found?
I think v.get("provisioned") will solve this issue as if "provisioned" is not found , value will be set to 'None'

**Review by cybernth — CHANGES_REQUESTED** [review]
https://github.com/Tendrl/ceph-integration/pull/214/files#r113143238

File: tendrl/ceph_integration/sds_sync/__init__.py (hunk 1 above)
Comment (GowthamShanmugam) [reply tendrl/ceph_integration/sds_sync/__init__.py]: @anmolsachan  @shtripat if provisioned field is not present then logic  written here also will give wrong value https://github.com/Tendrl/ceph-integration/blob/develop/tendrl/ceph_integration/sds_sync/__init__.py#L393

so setting 0 here is not solve the problem.

if provisioned filed is not there the du command will give used only, but as per loggic we written there will give wrong out put. so this solution will come there i feel.

Comment (shtripat) [reply tendrl/ceph_integration/sds_sync/__init__.py]: @GowthamShanmugam you can set `None` in sds_sync you can handle this `None` case properly. That would be the proper solution.

Comment (anmolsachan) [reply tendrl/ceph_integration/sds_sync/__init__.py]: I had given it 0 as here in definition, data type of provisioned shows int (https://github.com/Tendrl/ceph-integration/blob/develop/tendrl/ceph_integration/objects/definition/ceph.yaml#L271) . Should I put it as None or if has to be integer then we can also set -1, as provisioned size 0 is also a possible scenario ? @shtripat @nthomas-redhat @GowthamShanmugam

Comment (shtripat) [reply tendrl/ceph_integration/sds_sync/__init__.py]: @anmolsachan I feel just `v.get("provisioned")` you should try and it should work. The only thing in sds_sync take care properly for `None` case.

Comment (cybernth) [reply tendrl/ceph_integration/sds_sync/__init__.py]: There is another issue self._to_bytes('None') won't be right

**Review by r0h4n — APPROVED** [review]

**Review by r0h4n — APPROVED** [review]

**Pull request merged**